File: Prep-2025/prog/0016.cpp
#include<bits/stdc++.h>
using namespace std;
int n;
string s;
int Adrian=0;
int Bruno=0;
int Goran=0;
string a="ABC";
string b="BABC";
string g="CCAABB";
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cin>>n;
    cin>>s;
    for(int i=0;i<s.length();++i){
        if(a[i%3]==s[i])++Adrian;
        if(b[i%4]==s[i])++Bruno;
        if(g[i%6]==s[i])++Goran;
    }
    int mx=max({Adrian,Bruno,Goran});
    cout<<mx<<"\n";
    if(mx==Adrian)cout<<"Adrian\n";
    if(mx==Bruno)cout<<"Bruno\n";
    if(mx==Goran)cout<<"Goran\n";
    return 0;
}

File: python-prog/0016.py
import sys
n=int(sys.stdin.readline())
s=sys.stdin.readline()
l=len(s)
adrian="ABC"
bruno="BABC"
goran="CCAABB"
score_adrain=0
score_bruno=0
score_goran=0
for i in range(l):
  if s[i]==adrian[i%3]:
    score_adrain+=1
  if s[i]==bruno[i%4]:
    score_bruno+=1
  if s[i]==goran[i%6]:
    score_goran+=1
mx=max(score_adrain,score_bruno,score_goran)
print(mx)
if mx==score_adrain:
  print("Adrian")
if mx==score_bruno:
  print("Bruno")
if mx==score_goran:
  print("Goran")